src/services/raindrop/collection.ts

```diff
@@ -7,3 +18,52 @@ export const searchTerm = async (query: string, collectionId: string = "0") => {
 
   return httpClient.get(`/raindrops/${collectionId}?search=${query}`);
 };
+
+/**
+ * Creates a generator that returns a list of Raindrops that have been updated
+ * after the given `since` date.
+ *
+ * @param since A date, used to keep only the raindrops updated since this date.
+ * @param collectionId The id of the collection to fetch.
+ * @param params Additional query parameters
+ * @returns A generator that can be used to get each page of results.
+ *
+ * @example
+ * const raindrops = await createCollectionUpdatedSinceGenerator(
+ *   new Date("2020-01-01"),
+ *   "0"
+ * ).next();
+ * // raindrops is an object with a done and a value
+ * // the generator is done when there is no more content.
+ * // value is a list of Raindrops
+ */
+export async function* createCollectionUpdatedSinceGenerator(
+  since: Date,
+  collectionId: string = "0",
+  params?: {
+    perPage?: number;
+  }
+) {
+  if (!httpClient) {
+    throw new Error("Raindrop client not initialized");
+  }
+  let pageOffset = 0;
+
+  while (true) {
+    const response = await httpClient.get(
+      `/raindrops/${collectionId}?sort-lastUpdate&perpage=${
+        params?.perPage ?? 10
+      }&page=${pageOffset}`
+    );
+    const raindrops = (await response.json()) as CollectionResponse;
+    const raindropsUpdatedSince = (raindrops.items ?? []).filter(
+      (r) => new Date(r.lastUpdate) >= since
+    );
+
+    if (raindropsUpdatedSince.length === 0) break;
+    yield raindropsUpdatedSince.map(raindropTransformer);
+
+    pageOffset += 1;
```

Comment:
A potential opportunity to speed this up is to start the next request just before we yield. That way, by the time the generator is requested again, either the request is already complete or at least on it's way. As it stands, the generator is essentially just killing time between requests, which increases overall latency.